ras_backstage/authentication/token_decoder.py

```diff
@@ -0,0 +1,35 @@
+import logging
+
+from flask import abort
+import jwt
+
+from ras_backstage.authentication.uaa import get_uaa_public_key
+
+
+logger = logging.getLogger(__name__)
+
+USER_ID_CLAIM = "user_id"
+
+
+def decode_access_token(access_token, verify=True):
+
+    try:
+        decoded_jwt = jwt.decode(
+            access_token,
+            verify=verify,
+            algorithms=None,
+            key=get_uaa_public_key(),
+            audience='ras_backstage',
+            leeway=10,
+        )
+        return decoded_jwt
+    except jwt.DecodeError:
+        logger.error(f"Unable to decode token {access_token} - confirm the UAA public key is correct")
+        abort(500)
```

Comment:
I don't think we should be returning a 500 error from here. Ideally all responses should be as close as possible to the route definition (e.g. in sign_in.py), it makes it much easier to see what the possible responses are for a given endpoint.

I think it would be clearer if we separated the decoding from getting the user_id. The route can then decide what to do with the decode error.

Reply:
I'm not sure I agree with this comment. Abort 500 raises an InternalServerError which is handled by flasks internal error handling, so it'll forward to the standard error template. 

I could raise an error manually and deal with it in the error handler instead, but it achieves the same result.

Reply:
AFAIK for your first point - this could pollute the route definition quite a lot. Abort is seen as good practice in Flask and lets you reliably end a request and (as long as you have the correct error handlers in place) display the correct error message to the user.

Reply:
I'll remove the try except block in the token decoder and let it ripple up to the sign in page. It can abort from there. However that means @insacuri comment about error handling in the token decoder will have to be ignored.

Reply:
My comment was more about leaking Flask code into the token decoder. I was thinking of it more like we have a token decoder method that can raise an internal server error, which doesn't make sense to me.